Design/DesignDoc.md

```diff
@@ -0,0 +1,220 @@
+
+
+## 2. Top Questions to Resolve in Review
+
+_List the most important questions you have about your design, or things that you are still debating internally that you might like help working through._
+
+1. Can we search expenses by date? 
+2. Would date need to be a string rather than a ZoneDateTime
+3. Payments through this API is not in scope
+4. Does the budget model need more variables? 
+
+
+## 3. Use Cases
+
+_This is where we work backwards from the customer and define what our customers would like to do (and why). You may also include use cases for yourselves (as developers), or for the organization providing the product to customers._
+
+U1. As a user, I want to be able to view all my expenditures
+
+U2. As a user, I want to be able to view my expenditures based on their associated tag (EX. food, utilities, rent, personal, loan).
+
+U3. As a user, I want to be able to create a new expenditure and then add the expense plus the associated tag (Reason for expense).
+U4. As a user, I want to be able to update an expenditure’s information if changes arise.
+
+U5. As a user, I want to be able to view a list of all my expenditures sorted by highest to lowest expense.
+
+U6. As a user, I want to be able to view a list of all my expenditures sorted by date (newest - oldest)
+
+U6. As a user, I want to be able to get a specific day’s worth of expenses.
+
+U7. As a user, I want to be able to get a specific week’s worth of expenses.
+
+U8. As a user, I want to be able to get a specific month’s worth of expenses.
+
+U9. As a user, I want to be able to get a specific year’s worth of expenses.
+
+U10. As a user, I want to be able to view a graph or chart that depicts my expenses for a certain period in an informative manner.
+
+
+
+## 4. Project Scope
+
+_Clarify which parts of the problem you intend to solve. It helps reviewers know what questions to ask to make sure you are solving for what you say and stops discussions from getting sidetracked by aspects you do not intend to handle in your design._
+
+
+
+### 4.1. In Scope
+
+_Which parts of the problem defined in Sections 1 and 2 will you solve with this design? This should include the base functionality of your product. What pieces are required for your product to work?_
+
+* Adding, updating, and retrieving/viewing expenditure information
+* User specific data (multiple users allowed)
+* Retrieving expenditure information based on tags
+
+### 4.2. Out of Scope
+
+_Based on your problem description in Sections 1 and 2, are there any aspects you are not planning to solve? Do potential expansions or related problems occur to you that you want to explicitly say you are not worrying about now? Feel free to put anything here that you think your team can't accomplish in the unit, but would love to do with more time._
+
+* This API does not link with any card provider or bank provider so
```

Comment:
It looks like this section is still a work in progress, but the only things you need to list here are things that you've described in your use cases. You haven't said anything in them about linking with a provider or facilitating payments, so there's no need to list those as out of scope. As an example - after you flesh everything out you might decide that viewing charts/visuals about the expenses is too much to get done, so you'd leave it in the use cases but list it in this section as something that you're not planning to get to. This gives you a nice set of stretch goals that you can pick up if you have more time at the end.